File: src/models/action_box.rs
use time::OffsetDateTime;
use uuid::Uuid;

pub struct ActionBox {
    pub id: Uuid,
    pub name: String,
    pub description: Option<String>,
    pub content: String, // Markdown
    pub durarion: OffsetDateTime,
    pub created_at: OffsetDateTime,
    pub finished_at: OffsetDateTime,
}

impl ActionBox {}

File: src/models/actionBox.ts
import { Box } from './box'
import type {ChildBox} from "./childBox.ts";

export class ActionBox extends Box {
  childId: string // Uuid v7
  content?: string // Markdown
  duration: number // Ms

  constructor(
    name: string,
    description: string,
    child: ChildBox,
    duration: number,
    content?: string,
  ) {
    super(name, description)
    this.childId = child.id
    this.content = content
    this.duration = duration
  }

  create(): this {
    return { ...this }
  }
}
